sfaira/data/store/carts/utils.py

```diff
@@ -17,18 +17,41 @@ def split_batch(x: Union[Tuple, Dict]):
     """
     if isinstance(x, tuple):
         batch_dim = x[0][0].shape[0]
-        for i in range(batch_dim):
-            output = []
-            for y in x:
-                if isinstance(y, tuple):
-                    output.append(tuple([z.iloc[[i], :] if isinstance(z, pd.DataFrame) else z[i, :] for z in y]))
-                else:
-                    output.append(y.iloc[[i], :] if isinstance(y, pd.DataFrame) else y[i, :])
-            yield tuple(output)
+        if split_size == 1:
+            for i in range(batch_dim):
+                output = []
+                for y in x:
+                    if isinstance(y, tuple):
+                        output.append(tuple([z.iloc[[i], :] if isinstance(z, pd.DataFrame) else z[i, :] for z in y]))
+                    else:
+                        output.append(y.iloc[[i], :] if isinstance(y, pd.DataFrame) else y[i, :])
+                yield tuple(output)
+        else:
```

Comment:
If we can add the single organism batches like mentioned above we can get rid of this, right?

Reply:
Yes we could deprecate this again!